### PR TITLE
Molecular optimization tool

### PR DESCRIPTION
**Pull request by wverastegui**
This tool use the semiempirical  tight binding method (GFNn-xTB) implemented in the xTB package to perform molecular optimization.

### CODE REVIEW

File: tools/xtb/xtb_molecular_optimization.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<tool id="xtb_molecular_optimization" name="xtb molecular optimization" version="0.1.0+galaxy0" python_template_version="3.5" profile="21.05">
```
Comment (bgruening) [inline tools/xtb/xtb_molecular_optimization.xml:1]:
```suggestion
<tool id="xtb_molecular_optimization" name="xtb molecular optimization" version="0.1.0+galaxy0" profile="21.05">
```

File: tools/xtb/xtb_molecular_optimization.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        <organization
+            url="https://www.recetox.muni.cz/"
+            email="GalaxyToolsDevelopmentandDeployment@space.muni.cz"
+            name="RECETOX MUNI"/>
+    </creator>
+    <requirements>
+        <requirement type="package" version="6.6.1">xtb</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        ln ${coordinates} mol.${coordinates.ext} &&
```
Comment (bgruening) [inline tools/xtb/xtb_molecular_optimization.xml:22]: please single-quote all data and text parameters.

File: tools/xtb/xtb_molecular_optimization.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    <requirements>
+        <requirement type="package" version="6.6.1">xtb</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        ln '${coordinates}' 'mol.${coordinates.ext}' &&
+        xtb 'mol.${coordinates.ext}' --opt '${opt_level}' && 
+        cp 'xtbopt.${coordinates.ext}' '${output}'
+    ]]></command>
+    <inputs>
+        <param label="Atomic coordinates file" type="data" name="coordinates" format="xyz, coord" help="Initial atomic coordinates file to be optimized."/>
```
Comment (bgruening) [inline tools/xtb/xtb_molecular_optimization.xml:27]:
```suggestion
        <param label="Atomic coordinates file" type="data" name="coordinates" format="xyz,coord" help="Initial atomic coordinates file to be optimized."/>
```

File: tools/xtb/xtb_molecular_optimization.xml [hunk 4]
```diff
@@ -0,0 +1,73 @@
+<tool id="xtb_molecular_optimization" name="xtb molecular optimization" version="0.1.0+galaxy0" profile="21.05">
```
Comment (bgruening) [inline tools/xtb/xtb_molecular_optimization.xml:1]: should the version here maybe corresponds to the xtb version? If so can you maybe use a macro?

File: tools/xtb/.shed.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+owner: recetox
```
Comment (bgruening) [inline tools/xtb/.shed.yml:1]: Such a kind of shed.yml https://github.com/galaxyproject/tools-iuc/blob/main/tools/aldex2/.shed.yml

is enough. You don't need the include part etc ...

Comment (wverastegui) [reply tools/xtb/.shed.yml:3]: Updated (4d60285)

File: tools/xtb/xtb_molecular_optimization.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        xtb 'mol.${coordinates.ext}' --opt '${opt_level}' && 
+        cp 'xtbopt.${coordinates.ext}' '${output}'
+    ]]></command>
+    <inputs>
+        <param label="Atomic coordinates file" type="data" name="coordinates" format="xyz,coord" help="Initial atomic coordinates file to be optimized."/>
+        <param type="select" label="Optimization Levels" name="opt_level" help="Geometry optimization convergence levels." >
+            <option value="crude">crude</option>
+            <option value="tight">tight</option>
+            <option value="extreme">extreme</option>
+        </param>
```
Comment (hechth) [inline tools/xtb/xtb_molecular_optimization.xml:32]: Please add the additional options.

File: tools/xtb/xtb_molecular_optimization.xml (hunk 4 above)
Comment (hechth) [reply tools/xtb/xtb_molecular_optimization.xml:1]: Yeah, I agree - the version needs to be updated and it should be moved to a macro.

File: tools/xtb/xtb_molecular_optimization.xml (hunk 6 above)
Comment (wverastegui) [reply tools/xtb/xtb_molecular_optimization.xml]: Added opt levels options ([53c0ffa](https://github.com/wverastegui/galaxytools/commit/53c0ffa223abe64fdf14b43245e1637c1d9ef464))

File: tools/xtb/xtb_molecular_optimization.xml (hunk 4 above)
Comment (wverastegui) [reply tools/xtb/xtb_molecular_optimization.xml]: Macros created [7638ea6](https://github.com/RECETOX/galaxytools/pull/394/commits/7638ea623cbc28d6e4a2787a408b95fec22dc8a4) & [a9a6e65](https://github.com/RECETOX/galaxytools/pull/394/commits/a9a6e65c379ff56865bb1ed48f7b524f86e1d28a)

**Review by hechth — APPROVED** [review]

**Comment by hechth** [comment]
@bgruening this also shows again the symlink error with docker - which means that every tool that uses symlink will now fail in the CI as the CI uses biocontainers. Any idea how to fix this except for no longer using symlinks?

File: tools/xtb/xtb_molecular_optimization.xml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    <description>Semiempirical quantum mechanical molecular optimization method</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="creator"/>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">xtb</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        ln '${coordinates}' 'mol.${coordinates.ext}' &&
```
Comment (bgruening) [inline tools/xtb/xtb_molecular_optimization.xml:11]: ln -s  softlinks, not hardlinks

Comment (hechth) [reply tools/xtb/xtb_molecular_optimization.xml:11]: Thanks for your invaluable linux skills which I am so dearly lacking!

**Pull request merged**